### PR TITLE
fix(chutes): add graceful fallback for model parsing

### PR DESCRIPTION
**Pull request by hannesrudolph**
## Summary

Fixes an issue where the Chutes API returns an empty object `{}` at the end of their models array, which caused the entire Zod parse to fail and prevented all dynamic models from loading.

image

This change aligns the Chutes provider with the OpenRouter fallback pattern for more robust API response handling.

## Changes

- Made schema fields optional to allow graceful parsing
- Added graceful fallback: if schema validation fails, falls back to raw `response.data.data` (mirrors OpenRouter pattern)
- Added per-item validation to skip invalid entries (empty objects, missing `id`, missing `context_length`)
- Added `maxTokens` fallback calculation (`contextWindow * 0.2`) when `max_model_len` is missing
- Added 4 new test cases covering edge cases for invalid/malformed API responses

## Testing

- All 26 related tests pass (12 fetcher tests + 14 handler tests)
- Manually verified that previously missing dynamic-only models (e.g., `MiniMaxAI/MiniMax-M2`) now appear in the UI

----

> [!IMPORTANT]
> Fixes Chutes API model parsing by making schema fields optional and adding fallbacks for invalid data, with new test cases for edge cases.
> 
>   - **Behavior**:
>     - Makes all fields in `ChutesModelSchema` optional in `chutes.ts` for robust parsing of incomplete API responses.
>     - Adds fallback to raw `response.data.data` if schema validation fails, mirroring OpenRouter pattern.
>     - Skips invalid entries (empty objects, missing `id`, missing `context_length`) in `getChutesModels()`.
>     - Calculates `maxTokens` as `contextWindow * 0.2` if `max_model_len` is missing.
>   - **Testing**:
>     - Adds 4 new test cases in `chutes.spec.ts` for edge cases: empty objects, missing `id`, missing `context_length`, and mixed valid/invalid items.
>     - Confirms all 26 related tests pass, including fetcher and handler tests.
>     - Manually verified dynamic-only models now appear in the UI.
> 
> <sup>This description was created by </sup>[Ellipsis](https://www.ellipsis.dev?ref=RooCodeInc%2FRoo-Code&utm_source=github&utm_medium=referral)<sup> for 3daea7ff52cc3a89566618a5c2afed20afbc38f1. You can [customize](https://app.ellipsis.dev/RooCodeInc/settings/summaries) this summary. It will automatically update as commits are pushed.</sup>

### CODE REVIEW

**Comment by ghost** [comment]
> **Oroocle**  Clock &nbsp;&nbsp;[See task on Roo Cloud](https://app.roocode.com/cloud-jobs/JSGHLs9H?utm_source=github-comment&utm_medium=link&utm_campaign=github.pr.review.sync)

Re-review completed. Previously flagged item is resolved.

- [x] Guard `context_length`/`max_model_len` types when falling back to raw data, to avoid propagating non-number values into `ModelInfo`.

> <details>
> <summary>Previous reviews</summary>
> 
> - 3daea7f: [Review #1](https://app.roocode.com/cloud-jobs/PpviBCct?utm_source=previous-reviews&utm_medium=link&utm_campaign=github.pr.review)
> </details>
<sub>Mention @roomote in a comment to request specific changes to this pull request or [fix all unresolved issues](https://app.roocode.com/fix-issues?repo=RooCodeInc/Roo-Code&prNumber=10279&commentId=3683871719).</sub>

File: src/api/providers/fetchers/chutes.ts [hunk 1]
```diff
@@ -32,33 +35,49 @@ export async function getChutesModels(apiKey?: string): Promise<Record<string, M
...
-		} else {
-			console.error(`Error parsing Chutes models: ${JSON.stringify(parsed.error.format(), null, 2)}`)
+
+			const contextWindow = m.context_length
+			const maxModelLen = m.max_model_len
+
+			// Skip models without context window information
+			if (!contextWindow) {
+				continue
+			}
```
Comment (ghost) [inline src/api/providers/fetchers/chutes.ts]: When `ChutesModelsResponseSchema.safeParse()` fails you fall back to raw `response.data?.data`, but `context_length` and `max_model_len` are then effectively untyped. The current `if (!contextWindow)` check will accept non-number values (for example a string like `"128000"`) and propagate a non-number `contextWindow` into `ModelInfo`, which can break downstream consumers. Consider guarding for finite numbers and skipping entries that are not numeric.

```suggestion
			const contextWindow = typeof m.context_length === "number" && Number.isFinite(m.context_length) ? m.context_length : undefined
			const maxModelLen = typeof m.max_model_len === "number" && Number.isFinite(m.max_model_len) ? m.max_model_len : undefined

			// Skip models without valid context window information
			if (!contextWindow) {
				continue
			}
```

<sub>[Fix it with Roo Code](https://app.roocode.com/fix-issue/?repo=RooCodeInc/Roo-Code&prNumber=10279&fixId=9e62531d28) or mention @roomote and request a fix.</sub>
